### PR TITLE
feat: refactor Sdk, make compatible with .NET 8 and CPM

### PR DESCRIPTION
**Pull request by amis92**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 .ionide/
 App_Data/
 packages/
+package/
```
Comment (amis92) [inline .gitignore:6]: This is the path that CI pipeline uses as pack output.

File: src/Allegro.DotnetSdk/Allegro.DotnetSdk.csproj [hunk 2]
```diff
@@ -1,13 +1,21 @@
-<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.Build.NoTargets">
+
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
```
Comment (ppiwow-apay) [inline src/Allegro.DotnetSdk/Allegro.DotnetSdk.csproj:4]: why still 6.0? you mentioned that this version support 8 only, not get it

File: Directory.Build.props [hunk 3]
```diff
@@ -1,38 +1,22 @@
...
+    <Copyright>© Allegro. All rights reserved.</Copyright>
+    <Description>$(MSBuildProjectName)</Description>
+    <Tags>Allegro</Tags>
+    <PackageLicenseFile>LICENSE</PackageLicenseFile>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <UseArtifactsOutput>true</UseArtifactsOutput>
+  </PropertyGroup>
 
-    <PropertyGroup Label="Project Settings" >
```
Comment (ppiwow-apay) [inline Directory.Build.props:7]: why not needed? moved somewhere?

File: Directory.Build.targets [hunk 4]
```diff
@@ -1,13 +1,3 @@
 <Project>
```
Comment (ppiwow-apay) [inline Directory.Build.targets:1]: need clarification

File: src/Allegro.DotnetSdk/Sdk/Allegro.NET.Sdk.CSharp.targets
Comment (ppiwow-apay) [inline src/Allegro.DotnetSdk/Sdk/Allegro.NET.Sdk.CSharp.targets:1]: need clarification on difference of purpose o CSharp.props file and CSarp.targets

File: src/Allegro.DotnetSdk/Sdk/editor.globalconfig [hunk 5]
```diff
@@ -436,12 +436,3 @@ dotnet_diagnostic.MA0056.severity = suggestion
 
 # MA0074: Avoid implicit culture-sensitive methods
 dotnet_diagnostic.MA0074.severity = suggestion
-
```
Comment (ppiwow-apay) [inline src/Allegro.DotnetSdk/Sdk/editor.globalconfig:439]: moved somewhere or in net 8 something is different?

**Review by ppiwow-apay — APPROVED** [review]

File: src/Allegro.DotnetSdk/Allegro.DotnetSdk.csproj (hunk 2 above)
Comment (amis92) [reply src/Allegro.DotnetSdk/Allegro.DotnetSdk.csproj:4]: Sdk supports .NET SDK v8 only, but doesn't really care about TFM (so projects can still target `net6.0` but have to use SDK `8.0.100`+). It's because the CustomAfterBuildDirectoryProps hook was added only recently in the SDK.

File: src/Allegro.DotnetSdk.Tests/projects/Net8LibValid/Class1.cs
Comment (michalromanowskiap) [inline src/Allegro.DotnetSdk.Tests/projects/Net8LibValid/Class1.cs:1]: out of curiosity, why we need this?

File: src/Allegro.DotnetSdk/Sdk/Allegro.NET.Sdk.CSharp.targets
Comment (amis92) [reply src/Allegro.DotnetSdk/Sdk/Allegro.NET.Sdk.CSharp.targets:1]: `.props` are imported *before* a `.*proj` file, and `.targets` are imported *after* `.*proj` file. It is also important that `.targets` is imported by the `CustomAfterBuildDirectoryTargets` hook, which means that `Directory.Build.targets` was also imported, and most of the .NET SDK targets were also imported, so we can for example remove `Using` (implicit usings) elements added by the .NET SDK's `.props`.

In other words, things that happen in `.props` happen before `.*proj` so setting switch like `AllegroDotnetSdkEnable*=false` in project file will not work - has to be earlier, like Directory.Build.props. Switches for things in `.targets` can be set in project file.

File: src/Allegro.DotnetSdk.Tests/projects/Net8LibValid/Class1.cs
Comment (amis92) [reply src/Allegro.DotnetSdk.Tests/projects/Net8LibValid/Class1.cs:1]: These are test projects that import the SDK and are built by test runs, as a test case.

File: src/Allegro.DotnetSdk/Sdk/editor.globalconfig (hunk 5 above)
Comment (amis92) [reply src/Allegro.DotnetSdk/Sdk/editor.globalconfig:439]: I believe it was never used, as these IDs are for diagnostics from CezaryPiatek's package which is not referenced by our Sdk.

File: Directory.Build.props (hunk 3 above)
Comment (amis92) [reply Directory.Build.props:7]: What I removed were mostly defaults, those which weren't were moved to the actual project file, as appropriate.

File: Directory.Build.targets (hunk 4 above)
Comment (amis92) [reply Directory.Build.targets:1]: talked-through online
